test/integration/job/job_test.go

```diff
@@ -4127,13 +4127,20 @@ func validateIndexedJobPods(ctx context.Context, t *testing.T, clientSet clients
 	for _, pod := range pods.Items {
 		if metav1.IsControlledBy(&pod, jobObj) {
 			if pod.Status.Phase == v1.PodPending || pod.Status.Phase == v1.PodRunning {
-				ix, err := getCompletionIndex(&pod)
+				annotationIx, err := getCompletionIndex(pod.Annotations)
```

Comment:
The KEP is PodIndexLabel.. I think we should probably grab the labels of a pod rather than the annotations.

Reply:
Sorry I see below that you grabbing the label.